x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/AbstractScriptFieldType.java

```diff
@@ -198,6 +199,9 @@ protected final void checkAllowExpensiveQueries(QueryShardContext context) {
 
     @Override
     public ValueFetcher valueFetcher(QueryShardContext context, SearchLookup lookup, String format) {
+        if (lookup == null) {
+            return v -> Collections.emptyList();
+        }
```

Comment:
Another option here would be to pass the Fetch-phase search lookup through all the highlighting code to HighlightUtils.loadValues(), and I think we probably will want to do that eventually, but it's not really ready yet given our uncertainty around how that lookup is built and passed around anyway.

Reply:
I was going to ask why it is null, the supplier could throw unsupported operation exception and not be null, but it is no supplier in this case. I am not liking that much that it can be null, but I am not sure how we can avoid that in this scenario.